lib/events-reader.js

```diff
@@ -245,37 +229,18 @@ module.exports = function (harvesterApp) {
             }
         };
 
-        var checkpointUpdate = function(that,doc){
-            return harvesterApp.adapter.update('checkpoint', that.checkpoint.id, {ts: doc.ts})
-        };
-
-        var debouncedCheckpointUpdate = debounce(checkpointUpdate,parseInt(_.get(harvesterApp, 'options.eventsReaderDebounceWait'), 10) || 100,{leading: true});
-
         EventsReader.prototype.updateCheckpointAndReschedule = function (doc) {
             var that = this;
             if (doc != null) {
 
-                var regexProfileMessage = 'checkpoint regex to cl ' + doc.ns + ' ts ' +  doc.ts;
-                log.profile(regexProfileMessage);
                 var regexCheckpoint = new RegExp('.*\\.checkpoints$', 'i');
                 var matchCheckpoint = regexCheckpoint.test(doc.ns);
-                log.profile(regexProfileMessage);
 
                 if (!matchCheckpoint) {
-
-                    var profileMessage = 'updateCheckpointAndReshedule ' + doc.ts;
-
-                    debug('doc checkpoint ' + doc.ts, doc);
-                    log.profile(profileMessage);
                     debug('updating checkpoint with ts: ' + logTs(doc.ts));
+                    checkpointWriter.checkpointEvent.emit('newCheckpoint', that.checkpoint.id, doc);
```

Comment:
I think this is a perfect scenario to use an event driven approach, most of the Node.js communication is done by events and in my point of view there's no reason the to communication between events-reader and checkpointWriter be orchestrated.
As the name "events-reader" says it just read events and emmits and event with says "We have a new checkpoint" the listeneres will decide what will do with that checkpoint.
If we put a function call in "events-reader" to persist this checkpoint we are adding more responsability to events-reader, the responsability to write data.